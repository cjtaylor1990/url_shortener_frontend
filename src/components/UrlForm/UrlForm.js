import React from 'react';
import './UrlForm.css';

import UrlAdder from '../UrlAdder/UrlAdder.js';
import UrlList from '../UrlList/UrlList.js';

class UrlForm extends React.Component {

	render() {
    	return(
			<div data-testid="urlForm" className="urlForm">
				<UrlAdder />
				<UrlList />
			</div>
		);
	};

};

export default UrlForm;
