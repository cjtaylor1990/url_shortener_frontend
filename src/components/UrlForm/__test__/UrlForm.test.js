import React from 'react';
import ReactDOM from 'react-dom';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';

import UrlForm from '../UrlForm.js';

afterEach(cleanup);

describe('User accesses UrlForm component', () => {
    
    it('renders without crashing', () => {
        const div = document.createElement('div');
        
        ReactDOM.render(<UrlForm />, div);
        
        ReactDOM.unmountComponentAtNode(div);
    });

    it('matches snapshot', () => {
        const tree = renderer.create(<UrlForm/>).toJSON();
        
        expect(tree).toMatchSnapshot();
    });

    it('UrlForm contains UrlAdder component', () => {
        const { getByTestId } = render(<UrlForm />);
        
        const renderedUrlAdder = getByTestId('urlAdder');
        
        expect(renderedUrlAdder).toBeInTheDocument();
    });

    it('UrlForm contains UrlListComponent', () => {
        const { getByTestId } = render(<UrlForm />);

        const renderedUrlList = getByTestId('urlList');

        expect(renderedUrlList).toBeInTheDocument();
    });
    
});
