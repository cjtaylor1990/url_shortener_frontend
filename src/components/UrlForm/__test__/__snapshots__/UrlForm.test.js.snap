// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User accesses UrlForm component matches snapshot 1`] = `
<div
  className="urlForm"
  data-testid="urlForm"
>
  <div
    className="urlAdder"
    data-testid="urlAdder"
  >
    <input
      className="fullUrl"
      placeholder="Original URL"
      type="text"
    />
    <input
      className="shortUrl"
      placeholder="Shortened URL"
      type="text"
    />
    <button
      className="addUrl"
      type="submit"
    >
       Add URL 
    </button>
  </div>
  <div
    className="urlList"
    data-testid="urlList"
  />
</div>
`;
